From 176f85d0a98fd8382fef138f5bf2c5edfa863536 Mon Sep 17 00:00:00 2001
From: David Bremner <bremner@debian.org>
Date: Thu, 23 Oct 2014 23:24:18 +0200
Subject: [PATCH] turn off anti-aliasing for plot output

This is a bit ugly but it works around

libcairo bug 85373  / debian bug 766534

which otherwise breaks the racket build
---
 share/pkgs/plot-lib/plot/private/common/plot-device.rkt | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/share/pkgs/plot-lib/plot/private/common/plot-device.rkt b/share/pkgs/plot-lib/plot/private/common/plot-device.rkt
index 543082c..f101266 100644
--- a/share/pkgs/plot-lib/plot/private/common/plot-device.rkt
+++ b/share/pkgs/plot-lib/plot/private/common/plot-device.rkt
@@ -126,7 +126,7 @@
       (send dc set-origin
             (+ old-origin-x (* old-scale-x dc-x-min))
             (+ old-origin-y (* old-scale-y dc-y-min)))
-      (send dc set-smoothing 'smoothed)
+      #;(send dc set-smoothing 'smoothed)
       (send dc set-text-mode 'transparent)
       (when clipping-rect?
         (send dc set-clipping-rect 0 0 dc-x-size dc-y-size))
@@ -288,13 +288,13 @@
           (cond [(eq? pen-style 'transparent)
                  (send dc set-smoothing 'unsmoothed)
                  (send dc draw-polygon vs 0 0 'winding)
-                 (send dc set-smoothing 'smoothed)]
+                 #;(send dc set-smoothing 'smoothed)]
                 [else
                  (define old-pen (send dc get-pen))
                  (send dc set-pen transparent-pen)
                  (send dc set-smoothing 'unsmoothed)
                  (send dc draw-polygon vs 0 0 'winding)
-                 (send dc set-smoothing 'smoothed)
+                 #;(send dc set-smoothing 'smoothed)
                  (send dc set-pen old-pen)
                  (draw-lines/pen-style dc (cons (last vs) vs) pen-style)]))))
     
-- 
2.1.1

